engine/src/main/java/com/datatorrent/stram/engine/Node.java

```diff
@@ -485,6 +485,10 @@ protected void deactivateSinks()
 
   void checkpoint(long windowId)
   {
+    if (operator instanceof Operator.CheckpointNotificationListener) {
```

Comment:
The call is to notify of the imminent checkpoint. If there is no checkpoint, then why call the listener.